programs/client/TestHint.h

```diff
@@ -11,9 +11,35 @@
 namespace DB
 {
 
-/// Checks expected server and client error codes in testmode.
-/// To enable it add special comment after the query: "-- { serverError 60 }" or "-- { clientError 20 }".
-/// Also you can enable echoing all queries by writing "-- { echo }".
+/// Checks expected server and client error codes in --testmode.
+///
+/// The following comment hints are supported:
+///
+/// - "-- { serverError 60 }" -- in case of you are expecting server error.
+///
+/// - "-- { clientError 20 }" -- in case of you are expecting client error.
+///
+///   Remember that the client parse the query first (not the server), so for
+///   example if you are expecting syntax error, then you should use
+///   clientError not serverError.
+///
+/// Examples:
```

Comment:
I also consider the idea of this PR questionable,
but at least it adds valuable comments - let's merge!